src/array.rs

```diff
@@ -16,7 +16,8 @@ use super::{Error, IterError, Metadata, Miniconf, Peekable};
 use core::fmt::Write;
 
 /// An array that exposes each element through their [`Miniconf`](trait.Miniconf.html) implementation.
-pub struct Array<T, const N: usize>(pub [T; N]);
+#[derive(Clone, Copy, PartialEq, Eq, Debug, PartialOrd, Ord, Hash)]
+pub struct Array<T, const N: usize>([T; N]);
```

Comment:
Will this break if you try to supply a `T` that isn't e.g. `PartialEq`? I don't think we want to restrict what types the user can supply, as they may be out of the end-user's control.

Reply:
Rust playground agrees! Wonderful news. Thanks for clarifying.